src/backend/replication/walsender.c

```diff
@@ -2670,7 +2670,16 @@ InitWalSenderSlot(void)
 		}
 	}
 
-	Assert(MyWalSnd != NULL);
+	/*
+	 * neon: in vanilla this doesn't happen because walsenders register in
```

Comment:
Is it only PG 16 problem? If not, will you open PRs for the rest of versions?

Reply:
No, it affects all versions, I just typically first open one PR to get approval to rebase less.

Reply:
Makes sense, thanks